apiV2/controller/propertyController.js

```diff
@@ -0,0 +1,48 @@
+import Property from '../models/property';
+import { propertys } from '../data/data';
+import { Agent,User } from '../models/users';
+
+class PropertyController {
+  postProperty(req, res) {
+      const id = propertys.length + 1;
+      const { price, address, city, state, type, imageUrl } = req.body;
+      const { password,isAdmin,token, ...noA } = res.locals.user;
+      const property = new Property(id, noA, price, address, city, state, type, imageUrl); 
+      Agent.createProperty(property);
+      console.log(process.env.NODE_ENV)
+      return res.status(201).send({ status:201, property });
+    };
+
+  updateProperty(req, res) {
+    const { property } = res.locals;
+    const {price, address, city, state, type, imageUrl} = req.body
+    const advert = Agent.updateProperty(property,price, address, city, state, type, imageUrl);
+    res.status(200).send({status:200,property:advert});
+  }
+
+  markSold(req, res) {
+      const { property } = res.locals;
+      const advert =  Agent.markPropertySold(property);
+      res.status(200).send({status:200,property:advert});
+  }
+
+  deleteProperty(req, res) {
+    const { property } = res.locals
+    Agent.deleteProperty(property);
+    return res.status(200).send({ status:200, message: 'property deleted successfully' });
+  }
+
+  // eslint-disable-next-line consistent-return
```

Comment:
@swaibat , I don't want you to disable any Eslint flag.. if we are using Airbnb then we must abide by all the rules thereof. please take note of this in subsequent PRs